compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1505,6 +1505,11 @@ pub enum StatementKind<'tcx> {
     /// counter varible at runtime, each time the code region is executed.
     Coverage(Box<Coverage>),
 
+    /// Denotes a call to the intrinsic function copy_overlapping, where `src_dst` denotes the
+    /// memory being read from and written to(one field to save memory), and size
+    /// indicates how many bytes are being copied over.
+    CopyNonOverlapping(Box<CopyNonOverlapping<'tcx>>),
```

Comment:
I can't find where this variant is actually constructed.

Reply:
It's currently not being constructed at all, I was not sure where to add it in, as I'm not sure where the lowering from HIR to MIR is occurring, or if I should add it in THIR. Any advice here would be appreciated.

Reply:
You can lower the call to `copy_nonoverlapping` intrinsic in compiler/rustc_mir/src/transform/lower_intrinsics.rs.

Reply:
I've also added it in `compiler/rustc_codegen_ssa/src/mir/block.rs`, would it be necessary to add both?

Reply:
It should only be in `lower_intrinsics.rs`. You can then stop handling `copy_nonoverlapping` in `compiler/rustc_codegen_ssa/src/mir/block.rs` and instead only handle `StatementKind::CopyNonOverlapping`. In addition you can panic when finding `StatementKind::CopyNonOverlapping` during borrowck.

Reply:
Just for my own understanding, why only in lower_intrinsics? It seems that constructing it in block would reduce the amount of work that needs to be done?

Reply:
Because of the lowering in `lower_intrinsics`, the `copy_nonoverlapping` intrinsic isn't reachable anymore in `block.rs`.